integration_tests/run_pyspark_from_build.sh

```diff
@@ -104,6 +104,11 @@ else
     then
         TEST_TAGS="-m $TEST_TAGS"
     fi
+
+    # Set per-executor cores, if unspecified.
+    # This prevents per-thread allocations (like Parquet read buffers) from overwhelming the heap.
+    export PYSP_TEST_spark_executor_cores=${PYSP_TEST_spark_executor_cores:-'10'}
```

Comment:
why 10?  We have a few other places where we try to configure things for local mode already why are the number of executor cores out of sync with the LOCAL_PARALLEL or NUM_LOCAL_EXECS?

https://github.com/NVIDIA/spark-rapids/blob/09c2b3353aa5b8e1e2eed0995022b414166daed8/integration_tests/run_pyspark_from_build.sh#L297

On a side note are databricks tests being run in local mode and configured badly? In a regular databricks cluster will we also run into this type of a problem? If so this workaround feels very much like it is going in the wrong direction, like we need to really fix the underlying problem ASAP instead of trying to work around it.

Reply:
I did see that.  I figured we might consider lower parallelism for local-mode than cluster-mode, and that a more appropriate number might be suggested in the review.  I have verified that this works with `4`.